### DISCUSSION TITLE
Custom directory, automatically parse and splice when referencing svg, and use it as the name of svg

### DISCUSSION THREAD

**Discussion by lxccc812** [unocss/unocss#3388] [Q&A]
Now, I have a lot of svg icons, all placed in the `@/assets/svg` directory. It's difficult to manage. How should I configure it to use something like the following:
`@/assets/svg/base-info/xxxx.svg` When used:`<div class="i-svg-base-info-xxxx"></div>`
When using `@/assets/svg/ccc/ddd.svg`:`<div class="i-svg-ccc-ddd"></div>`
![image](https://github.com/unocss/unocss/assets/110974969/76113782-159d-4768-b434-0b42c7987e29)

In addition, I don’t really want such a configuration. In this case, the configuration will appear bloated and not very smart.
![image](https://github.com/unocss/unocss/assets/110974969/f065c023-f8c4-4da9-b135-6e360ee0545e)

**Comment by zyyv** [comment]
Not sure if upstream already supports it, it has nothing to do with uno.

**Reply by lxccc812** [reply]
Is it related to the upstream? Can it provide reading of svg files in the svg folder?

**Reply by zyyv** [reply]
I don't know the file structure of your svg directory. If you want to provide a minimal reproduction, the problem can be better located.

**Reply by lxccc812** [reply]
This has nothing to do with specific directories, let's say I have the following directory structure


assets
- svg
  - c.svg
  - base
    - a.svg
  - test
    - b.svg
  - dddd
    - eeeee.svg
 
I need to configure
```ts
# uno.config.ts
presetIcons({
    scale: 1.2,
    collections: {
        svg: FileSystemIconLoader('./assets/svg'),
        'svg-base': FileSystemIconLoader('./assets/svg/base'),
        'svg-test': FileSystemIconLoader('./assets/svg/test'),
    },
}),
```
can be used
`<div class="i-svg-c"></div>`
`<div class="i-svg-base-a"></div>`
`<div class="i-svg-test-b"></div>`


If I do not configure `'svg-dddd': FileSystemIconLoader('./assets/svg/dddd'),`, but automatically read all svg files under assets, I can also use `<div class="i - svg -dddd-eeee"></div>`
